src/SalePercentage.php

```diff
@@ -245,12 +245,25 @@ public static function getProductSalePercentage($product) {
    *
    * @param int $salePercentage
    *   The product sale percentage.
+   * @param \WC_Product $product
+   *   The product object.
    *
    * @return bool
    *   TRUE if the sale percentage flash bubble should be displayed.
    */
-  public static function checkDisplaySalePercentage($salePercentage) {
-    return (!is_single() && $salePercentage >= static::getMinimumSalePercentage()) || is_single();
+  public static function checkDisplaySalePercentage($salePercentage, \WC_Product $product) {
+    $is_category_eligible = FALSE;
+    $product_category_terms = get_the_terms($product->get_id(), 'product_cat');
+    $eligible_categories = get_option('_sale_percentage_eligible_product_categories');
+    if (!empty($eligible_categories)) {
+      foreach ($product_category_terms as $term) {
+        if (in_array($term->term_id, $eligible_categories)) {
+          $is_category_eligible = TRUE;
+          break;
+        }
+      }
+    }
```

Comment:
Hi, would you consider optimising this a little bit for performance and less code?
- `get_the_terms` will return full WP_Term objects, while we only need term ids, a better way would be to use the native WC_Product `get_category_ids()` method instead
- instead of looping and breaking loop on found result we can use `array_intersect`
- I have not tested this myself, please feel free to adapt if needed

```suggestion
    $product_category_ids = $product->get_categories_ids();
    $eligible_category_ids = get_option('_sale_percentage_eligible_product_categories') ?? [];
    $is_category_eligible = !empty(array_intersect($eligible_category_ids, $product_category_ids));
```

Reply:
@bogdanarizancu 
Updated in the last commit and we added some more conditions because there are some new changes requested.

`$product->get_categories_ids();` is not working so i used `$product_category_ids = wp_get_post_terms($product->get_id(), 'product_cat', ['fields' => 'ids']);`